experiment/secret-sync-controller/client/client.go

```diff
@@ -0,0 +1,48 @@
+package client
+
+import (
+	"context"
+	"flag"
+	"os"
+	"path/filepath"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"google.golang.org/api/option"
+
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func NewK8sClientset() *kubernetes.Clientset {
+	var kubeconfig *string
+	if home := os.Getenv("HOME"); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	return clientset
+}
+
+func NewSecretManagerClient(ctx context.Context) *secretmanager.Client {
+	// Create the client.
+	jsonPath := "/usr/local/google/home/shanefu/gcloud_key.json"
+	client, err := secretmanager.NewClient(ctx, option.WithCredentialsFile(jsonPath))
```

Comment:
The hardcode means I and others can't run or test this.  I would recommend using the default, and expecting the user/environment to provide the correct credentials.
```suggestion
	client, err := secretmanager.NewClient(ctx)
```

Reply:
You're right. That was me testing if using multiple service-accounts is possible.
Modified.

By calling `client, err := secretmanager.NewClient(ctx)` if I have a KSA with workload identity binding with GSA, does it use that KSA instead of looking for a key in GOOGLE_APPLICATION_CREDENTIALS?